EndlessClient/HUD/Controls/HudControlsFactory.cs

```diff
@@ -416,6 +416,7 @@ private ChatTextBox CreateChatTextBox()
             };
             chatTextBox.OnEnterPressed += (_, _) => _chatController.SendChatAndClearTextBox();
             chatTextBox.OnClicked += (_, _) => _chatController.SelectChatTextBox();
+            chatTextBox.OnTextChanged += (_, _) => _chatController.ClearAndWarnIfJailAndGlobal();
```

Comment:
I'm not familiar with this behavior, is this what the vanilla client does?

Reply:
It is yeah, I can get a video of it if you like. Although... Client does go one extra and appears to check the message box on a regular interval rather than on keypress...

For example, this implementation will allow endlessclient to type a global message in aeven, walk in to jail and the message will still be there. However, when they try to send, they will be prevented from being able to send the message anyway. 

In the vanilla client, if you type a global message then walk in to jail, the client has a small delay and removes the message within a couple hundred microseconds it seems. 

The other thing I could add to this pr is this same chatbox check and run that check on map warp too.

Reply:
Keeping parity with the original client is important to me for this project, so as long as the original client does it, I'm fine with it. I just haven't played around with the behavior and wasn't able to test it when I was reviewing.

For the map change chat box clear, that would be nice to have. You can take a look at `MapChangedActions`, that's probably where you'd want to add it if the map is the jail map.

Reply:
this is vanilla with some trash quality https://youtu.be/0qw7OS7Pi_c